src/Workspaces/Remote/ServiceHub/Services/Initialization/RemoteInitializationService.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote;
+
+internal sealed class RemoteInitializationService(
+    BrokeredServiceBase.ServiceConstructionArguments arguments)
+    : BrokeredServiceBase(arguments), IRemoteInitializationService
+{
+    internal sealed class Factory : FactoryBase<IRemoteInitializationService>
+    {
+        protected override IRemoteInitializationService CreateService(in ServiceConstructionArguments arguments)
+            => new RemoteInitializationService(arguments);
+    }
+
+    /// <summary>
+    /// Remote API.
+    /// </summary>
+    public async ValueTask<int> InitializeAsync(WorkspaceConfigurationOptions options, string localSettingsDirectory, CancellationToken cancellationToken)
+    {
+        await RemoteExportProvider.InitializeAsync(localSettingsDirectory, cancellationToken).ConfigureAwait(false);
```

Comment:
move into RunServiceAsync?

Reply:
Adding a comment why that's a bad idea.

Reply:
It might be worth expanding on that comment if the reason it's broken is because something in RunServiceAsync depends on it...